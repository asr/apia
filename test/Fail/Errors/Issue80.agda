-- Issue #80.

-- The disjunction data type.
data _∨_ (A B : Set) : Set where
  inj₁ : A → A ∨ B
  inj₂ : B → A ∨ B

-- A different symbol for disjunction.
_⊕_ : Set → Set → Set
P ⊕ Q = P ∨ Q
{-# ATP definition _⊕_ #-}

postulate
  P Q : Set
  ⊕-comm : P ⊕ Q → Q ⊕ P
{-# ATP prove ⊕-comm #-}

-- The previous error was:

-- $ apia Issue80.agda
-- An internal error has occurred. Please report this as a bug.
-- Location of the error: src/Apia/Translation/Functions.hs:188

-- The current error is:

-- $ apia Issue80.agda
-- apia: the translation of ‘IssueXX._⊕_’ failed because it is not a FOL-definition
