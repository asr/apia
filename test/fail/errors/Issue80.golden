ret > ExitFailure 1
err > apia: the translation of ‘Issue80._⊕_’ failed because it is not a FOL-definition
err >
